airbyte-server/src/test/java/io/airbyte/server/handlers/DestinationDefinitionsHandlerTest.java

```diff
@@ -68,7 +77,9 @@ void setUp() {
     dockerImageValidator = mock(DockerImageValidator.class);
     destination = generateDestination();
     schedulerJobClient = spy(CachingSchedulerJobClient.class);
-    destinationHandler = new DestinationDefinitionsHandler(configRepository, dockerImageValidator, uuidSupplier, schedulerJobClient);
+    webServer = new MockWebServer();
```

Comment:
I'm not super convinced we need this. Would like to figure out why mocking wasn't working and see if we can fix that. If it's truly not possible then we can use this, but if mocking can work I'd prefer that.

e.g. this worked.
```
@Test
void testHttpClient() {
  final HttpResponse httpResponse = mock(HttpResponse.class);
  when(httpResponse.body()).thenReturn("hello");
  System.out.println("httpResponse.body() = " + httpResponse.body());
}
```

Reply:
Let me spend half an hour poking around to see if I can replicate this.

Reply:
Some of my prev annoyance with mocking was from the previous async call and having to do future -> httpresponse -> body.

Yes, you are right. Managed to get the mock stuff to work. Not sure what I was running into previously..

I still have a slight preference for using `MockWebServer`. I quite like how mocking is minimal and the control we get, especially on headers and all.

This is what it'll look timeout test will look like with pure mockito. Thought it's concise, I don't like how the delay statement has so much in it haha.
```    var mockHttpClient = mock(HttpClient.class);
    var mockResponse = mock(HttpResponse.class);
    when(mockResponse.body()).thenReturn("whooo");
    
    doAnswer( new AnswersWithDelay( 1000,  new Returns(mockResponse)) ).when(mockHttpClient).send(any(), any());
    
    var request = HttpRequest.newBuilder().uri(URI.create("https://www.google.com")).build();
    var send = mockHttpClient.send(request, BodyHandlers.ofString());

    System.out.println(send.body());
    ```

Definitely personal opinion, happy to change everything to mock if you feel strongly.

Reply:
kk. let's try it as is and see how it goes. i think my current reservation is that i don't know what mockwebserver is doing under the hood to make this all work. it seems like we're introducing some magic into the stack. we can change it in the future if it poses problems.